src/core/include/openvino/op/util/avg_pool_base.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/attr_types.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+class OPENVINO_API AvgPoolBase : public Op {
+public:
+    OPENVINO_OP("AvgPool", "util");
+    AvgPoolBase() = default;
+
+    /// \brief      Constructs a batched average pooling operation.
+    ///
+    /// \param      arg            The output producing the input data batch tensor.<br>
+    ///                            `[d1, dn]`
+    /// \param      strides        The strides.<br> `[n]`
+    /// \param      pads_begin     The beginning of padding shape.<br> `[n]`
+    /// \param      pads_end       The end of padding shape.<br> `[n]`
+    /// \param      kernel         The kernel shape.<br> `[n]`
+    /// \param      exclude_pad    If false then averages include padding elements, each
+    ///                            treated as the number zero.  If true, padding
+    ///                            elements
+    ///                            are entirely ignored when computing averages.
+    /// \param      rounding_type  Whether to use ceiling or floor rounding type while
+    ///                            computing output shape.
+    /// \param      auto_pad       Padding type to use for additional padded dimensions
+    AvgPoolBase(const Output<Node>& arg,
+                const Strides& strides,
+                const Shape& pads_begin,
+                const Shape& pads_end,
+                const Shape& kernel,
+                bool exclude_pad,
+                op::RoundingType rounding_type = op::RoundingType::FLOOR,
+                const PadType& auto_pad = op::PadType::EXPLICIT);
+
+    void validate_and_infer_types() override;
+    bool visit_attributes(AttributeVisitor& visitor) override;
```

Comment:
I suggest adding a similar method to the `MaxPoolBase` class with common attributes for v1, v8 and v14, as I did for `AvgPoolBase`

For example for `MaxPool-8`:
```
bool MaxPool::visit_attributes(AttributeVisitor& visitor) {
    OV_OP_SCOPE(v8_MaxPool_visit_attributes);
    visitor.on_attribute("index_element_type", m_index_element_type);
    visitor.on_attribute("axis", m_axis);
    return util::MaxPoolBase::visit_attributes(visitor);
}
```

Reply:
Good idea, let's have it in a separate PR.

Reply:
Let's also do it in a separate Good First Issue. @Vladislav-Denisov would you like to create one or do you prefer I'd do it?

Reply:
@p-wysocki Do you mean to create new GoodFirstIssue or only new PR?

I would be glad to create new PR to update `MaxPoolBase` implementation 😃

Reply:
PR - certainly, Good First Issue - if you feel like it you can create one, if not, I'll do it, just let me know. :)

Reply:
I can create both a PR and a GoodFirstIssue.
Thanks!